nuttx-patches/priority_restoration_fix.patch

```diff
@@ -411,14 +411,14 @@ index 6b9f05a..bc5c186 100644
 +  if (sem->hhead != NULL)
      {
        serr("ERROR: Semaphore destroyed with holders\n");
-+      DEBUGASSERT(sem->hhead != NULL);
++      DEBUGASSERT(sem->hhead == NULL);
        (void)sem_foreachholder(sem, sem_recoverholders, NULL);
      }
  #else
 -  if (sem->holder.htcb)
-+  if (sem->holder[0].htcb != NULL || sem->holder[0].htcb != NULL)
++  if (sem->holder[0].htcb != NULL || sem->holder[1].htcb != NULL)
      {
-+      DEBUGASSERT(sem->holder[0].htcb != NULL || sem->holder[0].htcb != NULL);
++      DEBUGASSERT(sem->holder[0].htcb == NULL || sem->holder[1].htcb == NULL);
```

Comment:
@bkueng - Thank you.  Upstream has already changed this to DEBUGPANIC(); - I will factor this in.